src/app/screens/ConfirmKeysend/index.tsx

```diff
@@ -6,25 +6,25 @@ import PublisherCard from "@components/PublisherCard";
 import SuccessMessage from "@components/SuccessMessage";
 import { useState, MouseEvent, useRef, useEffect } from "react";
 import { useTranslation } from "react-i18next";
-import { useNavigate, useSearchParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
 import { toast } from "react-toastify";
 import ScreenHeader from "~/app/components/ScreenHeader";
 import { useSettings } from "~/app/context/SettingsContext";
+import { useNavigationState } from "~/app/hooks/useNavigationState";
 import { USER_REJECTED_ERROR } from "~/common/constants";
 import msg from "~/common/lib/msg";
 import utils from "~/common/lib/utils";
 import { getFiatValue } from "~/common/utils/currencyConvert";
 import getOriginData from "~/extension/content-script/originData";
-import type { OriginData } from "~/types";
 
-type Props = {
-  origin?: OriginData;
-  destination?: string;
-  customRecords?: Record<string, string>;
-  valueSat?: string;
-};
+function ConfirmKeysend() {
+  const navState = useNavigationState();
+  const destination = navState.args?.destination as string;
+  const amount = navState.args?.amount as string;
+  const customRecords = navState.args?.customRecords as Record<string, string>;
+  const origin = navState.origin;
```

Comment:
Do we know why we use `origin` and `originRef` here? Wondering if that's needed. Looks like this happened by accident earlier? What do you think?